spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,579 @@
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.collections.CollectionUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+@SpecTest
+class ExportSpecTest {
+    private static final String EXPORT_JOB_KEY = "ExportJob";
+    private static final String RECORD_CLEANER_JOB_KEY = "ImportRecordCleanerJob";
+    private static final String UNDO_IMPORTS_JOB_KEY = "UndoImportsJob";
+    /** The following are directory location paths within a manifest */
+    private static final String EXPORT_PATH = "export";
+    private static final String PRODUCTS_PATH = "export/products";
+    private static final String ENTITILEMENTS_PATH = "export/entitlements";
+    private static final String ENTITILEMENT_CERTIFICATES_PATH =  "export/entitlement_certificates";
+    private static final String DISTRIBUTOR_VERSION_PATH = "export/distributor_version";
+    private static final String CONSUMER_TYPE_PATH = "export/consumer_types";
+    private static final String CDN_PATH = "export/content_delivery_network";
+    private static final String RULES_PATH = "export/rules2";
+
+    @Test
+    @DisplayName("should allow manifest creation with read only user")
+    void shouldAllowManifestCreationWithReadOnlyUser() throws Exception {
+        ApiClient client = ApiClients.admin();
+        OwnerDTO owner = client.owners().createOwner(Owners.random());
+        ConsumerDTO readOnlyConsumer = createReadOnlyConsumer(client, owner);
+
+        File readOnlyManifest = createExport(client, readOnlyConsumer.getUuid(), null);
+        ZipFile readOnlyExport = ExportUtil.getExportArchive(readOnlyManifest);
+        List<ZipEntry> exportDir = ExportUtil.getFilesInDirectory(EXPORT_PATH, readOnlyExport);
+        List<ZipEntry> consumerFiles = exportDir.stream()
+            .filter(e -> e.getName().contains("consumer.json"))
+            .collect(Collectors.toList());
+        assertEquals(1, consumerFiles.size());
+
+        Map<String, File> actual = ExportUtil.copyToTempFiles(readOnlyExport, consumerFiles);
+        JsonNode root = new ObjectMapper()
```

Comment:
The issue that I run into is that most of the json files in the manifest can't be de-serialized to the object classes because there are unrecognized fields in the json files. I included the error message that is thrown when I try to de-serialize into the json file as a ConsumerDTO class.

```
  com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "created" (class org.candlepin.dto.api.v1.NestedOwnerDTO), not marked as ignorable (5 known properties: "contentAccessMode", "id", "displayName", "key", "href"])
   at [Source: (ZipFileInflaterInputStream); line: 1, column: 100] (through reference chain: org.candlepin.dto.api.v1.ConsumerDTO["owner"]->org.candlepin.dto.api.v1.NestedOwnerDTO["created"])
      at app//org.candlepin.spec.ExportSpecTest.shouldAllowManifestCreationWithReadOnlyUser(ExportSpecTest.java:123)
```

Reply:
It looks like setting the ObjectMapper DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES configuration to false resolves that issue!